zomes/wiki/code/src/page.rs

```diff
@@ -95,75 +84,95 @@ pub fn page_def() -> ValidatingEntryType {
-    timestamp: u64,
+    timestamp: String,
 ) -> ZomeApiResult<String> {
-    let page_address = create_page_if_non_existent(title.clone())?;
-    let sections: Vec<Address> = sections
-        .into_iter()
-        .map(|element| {
-            hdk::api::commit_entry(&Section::from(element, page_address.clone()).entry())
-        })
-        .filter_map(Result::ok)
-        .collect();
+    let anchor_address = holochain_anchors::anchor("wiki_pages".to_string(), title.clone())?;
 
-    hdk::api::update_entry(
-        Page::from(title.clone(), sections, Some(timestamp)).entry(),
-        &page_address,
+    create_page_if_non_existent(
+        title.clone(),
+        sections
+            .into_iter()
+            .map(|section| hdk::api::commit_entry(&section.from(anchor_address.clone()).entry()))
+            .filter_map(Result::ok)
+            .collect(),
+        timestamp,
     )?;
 
     Ok(title)
 }
-pub fn update_page(sections: Vec<Address>, title: String, timestamp: u64) -> ZomeApiResult<String> {
-    let page_address = create_page_if_non_existent(title.clone())?;
+pub fn update_page(
+    sections: Vec<Address>,
+    title: String,
+    timestamp: String,
+) -> ZomeApiResult<String> {
+    let page_address =
+        create_page_if_non_existent(title.clone(), sections.clone(), timestamp.clone())?;
     hdk::api::update_entry(
-        Page::from(title.clone(), sections, Some(timestamp)).entry(),
+        Page::from(title.clone(), sections, timestamp).entry(),
         &page_address,
     )?;
     Ok(title)
 }
-pub fn get_page(title: String) -> ZomeApiResult<JsonString> {
-    utils::get_entry(Page::initial(title).entry().address())
-}
-
-pub fn get_titles() -> ZomeApiResult<Vec<String>> {
-    let anchor_address = utils::anchor("wiki_pages", "all_pages")?;
+pub fn get_page(title: String) -> ZomeApiResult<Page> {
     Ok(hdk::utils::get_links_and_load_type::<Page>(
-        &anchor_address,
-        LinkMatch::Exactly("anchor->Page".into()),
+        &holochain_anchors::anchor("wiki_pages".into(), title)?,
+        LinkMatch::Exactly("anchor->Page"),
         LinkMatch::Any,
-    )?
-    .into_iter()
-    .map(|page| page.title)
-    .collect())
+    )?[0]
+        .clone())
+}
+pub fn get_titles() -> ZomeApiResult<Vec<String>> {
+    get_titles_filtered("".to_string())
 }
-
 pub fn get_titles_filtered(data: String) -> ZomeApiResult<Vec<String>> {
-    let anchor_address = utils::anchor("wiki_pages", "all_pages")?;
-    Ok(hdk::utils::get_links_and_load_type::<Page>(
-        &anchor_address,
-        LinkMatch::Exactly("anchor->Page".into()),
-        LinkMatch::Regex(&("^".to_owned() + &data)),
-    )?
-    .into_iter()
-    .map(|page| page.title)
-    .collect())
+    let anchor_address = Entry::App(
+        holochain_anchors::ANCHOR_TYPE.into(),
+        holochain_anchors::Anchor {
+            anchor_type: "wiki_pages".to_string(),
+            anchor_text: None,
+        }
+        .into(),
+    )
+    .address();
+    Ok(
```

Comment:
For readibility, try not to put this much code on the same statement. Split this into different lines.